crypto/hmac/hmac.c

```diff
@@ -37,7 +37,8 @@ int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
     if (key != NULL) {
         reset = 1;
         j = EVP_MD_block_size(md);
-        OPENSSL_assert(j <= (int)sizeof(ctx->key));
+        if (!ossl_assert(j <= (int)sizeof(ctx->key)))
+            goto err;
```

Comment:
Both values are essentially constant here, one is compile-time constant and another can reside in read-only segment. In other words outcome can't be affected by user input, which is the criteria for passing control to the caller. I think it should be removed (or degraded to "soft" assert), because it's compile-time condition...